src/Core/Compiler/CompilerService.cs

```diff
@@ -132,6 +132,7 @@ public IEnumerable<QsNamespaceElement> IdentifyElements(string source)
             var loadOptions = new CompilationLoader.Configuration
             {
                 GenerateFunctorSupport = true,
+                LoadReferencesBasedOnGeneratedCsharp = true,
```

Comment:
I think this should depend on whether compilation is done for an execution target or not.

Reply:
Yes, good point, that makes sense to me as well.